app/models/manageiq/providers/google/cloud_manager/event_catcher/stream.rb

```diff
@@ -2,6 +2,14 @@
 # event type messages. Parsing of the message is deferred to the caller of
 # #poll.
 class ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream
+  # Generic exception if we are unable to reach GCP
+  class ProviderUnreachable < ManageIQ::Providers::BaseManager::EventCatcher::Runner::TemporaryFailure
```

Comment:
This seems very generic, I wonder if this should be bubbled up to a more generic parent class eventually.